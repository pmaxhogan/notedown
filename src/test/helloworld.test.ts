import { mount } from "@vue/test-utils";
import HelloWorld from "../components/HelloWorld.vue";
import { expect, test } from "vitest";

test("hello world component", async () => {
  expect(HelloWorld).toBeTruthy();

  const wrapper = mount(HelloWorld, {
    props: {
      msg: "test msg",
    },
  });

  expect(wrapper.text()).toContain("test msg");
  expect(wrapper.html()).toMatchSnapshot();

  // check that the .green class is applied to msg
  expect(wrapper.find(".green").exists()).toBe(true);
  expect(wrapper.find(".green").text()).toBe("test msg");
});
