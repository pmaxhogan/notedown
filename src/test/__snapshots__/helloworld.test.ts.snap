// Vitest Snapshot v1

exports[`hello world component 1`] = `
"<div data-v-e17ea971=\\"\\" class=\\"greetings\\">
  <h1 data-v-e17ea971=\\"\\" class=\\"green\\">test msg</h1>
  <h3 data-v-e17ea971=\\"\\"> Youâ€™ve successfully created a project with <a data-v-e17ea971=\\"\\" href=\\"https://vitejs.dev/\\" target=\\"_blank\\" rel=\\"noopener\\">Vite</a> + <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/\\" target=\\"_blank\\" rel=\\"noopener\\">Vue 3</a>. </h3>
</div>"
`;
